test/client.test.js

```diff
@@ -123,6 +123,38 @@ describe('Client', () => {
     await client.ping()
     expect(fetch).toBeCalled()
   })
+
+  test('set queryTimeout on client instance', () => {
+    client.queryTimeout(1000)
+    expect(client._queryTimeout).toEqual(1000)
+
+    client.queryTimeout(2000)
+    expect(client._queryTimeout).toEqual(2000)
+  })
+
+  test.only('properly sets query timeout header', async () => {
+    const fetch = jest.fn((expr, opts) => {
+      expect(opts.headers['X-Query-Timeout']).toEqual(2000)
+
+      return Promise.resolve({
+        headers: new Set(),
+        text: () =>
+          Promise.resolve(
+            JSON.stringify({
+              data: ['Collection("my_collection")'],
+            })
+          ),
+      })
+    })
+
+    const testClient = util.getClient({
+      fetch,
+    })
+
+    await testClient.query(query.Paginate(query.Collections()), {
+      queryTimeout: 2000,
+    })
+  })
```

Comment:
Maybe are you missing an `expect`? I think you can check something like `expect(fetch).wasCalledWith(...)`

Reply:
@BrunoQuaresma I have an `expect` on line 137 so I can make sure our `X-Query-Timeout` header is going out properly with the request. Do you think that is enough to cover this functionality?